examples/optsCommands.c

```diff
@@ -0,0 +1,161 @@
+		      keyNew (SPEC_BASE_KEY "/printversion", KEY_META, "description",
+			      "print version information and exit (ignoring all other options/commands/parameters)", KEY_META, "opt", "v",
+			      KEY_META, "opt/arg", "none", KEY_META, "opt/long", "version", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/getter", KEY_META, "description", "get a key's value", KEY_META, "command", "get", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/getter/verbose", KEY_META, "description",
+			      "print additional information about where the value comes from", KEY_META, "opt", "v", KEY_META, "opt/long",
+			      "verbose", KEY_META, "opt/arg", "none", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/getter/keyname", KEY_META, "description", "name of the key to read", KEY_META, "args",
+			      "indexed", KEY_META, "args/index", "0", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter", KEY_META, "description", "set a key's value", KEY_META, "command", "set", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter/verbose", KEY_META, "description",
+			      "print additional information about where the value will be stored", KEY_META, "opt", "v", KEY_META,
+			      "opt/long", "verbose", KEY_META, "opt/arg", "none", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter/keyname", KEY_META, "description", "name of the key to write", KEY_META, "args",
+			      "indexed", KEY_META, "args/index", "0", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter/value", KEY_META, "description", "value to be written", KEY_META, "args", "indexed",
+			      KEY_META, "args/index", "1", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/dynamic/#", KEY_META, "description", "dynamically call a user-supplied command", KEY_META,
+			      "args", "remaining", KEY_END),
+		      KS_END);
+}
+
+int main (int argc, const char ** argv)
+{
+	KeySet * ks = createSpec ();
+	Key * errorKey = keyNew (BASE_KEY, KEY_END);
+
+	int result = elektraGetOpts (ks, argc, argv, (const char **) environ, errorKey);
+	if (result == -1)
+	{
+		fprintf (stderr, "ERROR: %s\n", keyString (keyGetMeta (errorKey, "error/reason")));
+		keyDel (errorKey);
+		ksDel (ks);
+		return EXIT_FAILURE;
+	}
+
+	if (result == 1)
+	{
+		char * help = elektraGetOptsHelpMessage (errorKey, NULL, NULL);
+		fprintf (stderr, "%s\n", help);
+		elektraFree (help);
+		keyDel (errorKey);
+		ksDel (ks);
+		return EXIT_SUCCESS;
+	}
+
+	printf ("A real implementation would now \n");
+
+	Key * lookup = ksLookupByName (ks, BASE_KEY "/printversion", 0);
+	if (lookup != NULL && elektraStrCmp (keyString (lookup), "1") == 0)
+	{
+		printf ("print version information\n");
+
+		keyDel (errorKey);
+		ksDel (ks);
+
+		return EXIT_SUCCESS;
+	}
+
+	lookup = ksLookupByName (ks, BASE_KEY "", 0);
+	if (lookup != NULL && elektraStrCmp (keyString (lookup), "getter") == 0)
+	{
+		lookup = ksLookupByName (ks, BASE_KEY "/getter/keyname", 0);
+		if (lookup == NULL || strlen (keyString (lookup)) == 0)
+		{
+			printf ("report the error 'missing parameter: keyname'\n");
```

Comment:
how can I come here? When I execute `bin/optsCommands get`, already the ERROR: path above will be executed (ERROR: Expected at least 1 non-option arguments, but only got 0).

Reply:
the `lookup == NULL` should never be true AFAIK know, but `strlen (keyString (lookup)) == 0` is true, if you call `bin/optsCommands get ""`

Reply:
```suggestion
			printf ("report the error 'empty parameter: keyname'\n");
```